MonoGame.Framework/Graphics/GraphicsDevice.cs

```diff
@@ -843,6 +844,29 @@ public void DrawIndexedPrimitives(PrimitiveType primitiveType, int baseVertex, i
             PlatformDrawIndexedPrimitives(primitiveType, baseVertex, startIndex, primitiveCount);
         }
 
+        public void DrawIndexedPrimitives(PrimitiveType primitiveType, int baseVertex, int startIndex, int primitiveCount)
+        {
+            if (_vertexShader == null)
+                throw new InvalidOperationException("Vertex shader must be set before calling DrawIndexedPrimitives.");
+
+            if (_vertexBuffer == null)
+                throw new InvalidOperationException("Vertex buffer must be set before calling DrawIndexedPrimitives.");
+
+            if (_indexBuffer == null)
+                throw new InvalidOperationException("Index buffer must be set before calling DrawIndexedPrimitives.");
+
+            if (primitiveCount <= 0)
+                throw new ArgumentOutOfRangeException("primitiveCount");
+
+            unchecked
+            {
+                _graphicsMetrics._drawCount++;
+                _graphicsMetrics._primitiveCount += (ulong)primitiveCount;
+            }
+
+            PlatformDrawIndexedPrimitives(primitiveType, baseVertex, startIndex, primitiveCount);
+        }

```

Comment:
I'm fine with adding a new overload which removes the confusing and unused parameter.

We should go a step further and change the original method and rename `minVertexIndex` to `unused` with the documentation description of  `This is unused and remains here only for XNA API compatibility`.

We will need to keep the original method as long as we want to maintain XNA API compatibility, but internally the code should just forward the call to the new method.

Still lets see what @KonajuGames thinks before proceeding on any of this.